Documentation/contributing/development/hive.rst

```diff
@@ -780,6 +780,186 @@ Few guidelines one should strive to follow when implementing larger cells:
+interacting with it. We can now define our test runner:
+
+.. literalinclude:: ../../../contrib/examples/script/example_test.go
+   :caption: contrib/examples/script/example_test.go
+   :language: go
+   :tab-width: 4
+
+And with the test runner in place we can now write our test script:
+
+.. literalinclude:: ../../../contrib/examples/script/testdata/example.txtar
+   :caption: contrib/examples/script/testdata/example.txtar
+   :language: shell
+
+With everything in place we can now run the tests:
+
+.. code-block:: shell-session
+
+  $ cd contrib/examples/script 
+  $ go test .
+  === RUN   TestScript
+  === RUN   TestScript/example.txtar
+    scripttest.go:251: 2025-02-26T08:32:25Z
+    scripttest.go:253: $WORK=/tmp/TestScriptexample.txtar2477299450/001
+    scripttest.go:72:
+        DATADIR=/home/jussi/go/src/github.com/cilium/cilium/contrib/examples/script/testdata
+        PWD=/tmp/TestScriptexample.txtar2477299450/001
+        WORK=/tmp/TestScriptexample.txtar2477299450/001
+        TMPDIR=/tmp/TestScriptexample.txtar2477299450/001/tmp
+
+    scripttest.go:72: #! --enable-example=true
+        # ^ an (optional) shebang can be used to configure cells
+        # This is a comment that starts a section of commands (0.000s)
+        > echo 'hello'
+        [stdout]
+        hello
+    logger.go:256: level=INFO msg="Starting hive"
+    logger.go:256: level=INFO msg="Started hive" duration=1.53µs
+    scripttest.go:72: # The test hive has not been started yet, let's start it! (0.000s)
+        > hive/start
+    logger.go:256: level=INFO msg="SayHello() called" module=example name=foo greeting=Hello,
+    scripttest.go:72: # Cells can provide custom commands (0.000s)
+        > example/hello foo
+        calling SayHello(foo, Hello,)
+        [stdout]
+        Hello, foo
+        > stdout 'Hello, foo'
+        matched: Hello, foo
+    scripttest.go:72: # Check that call count equals 1 (0.000s)
+        > example/counts
+        [stdout]
+        1 SayHello()
+        > stdout '1 SayHello()'
+        matched: 1 SayHello()
+    scripttest.go:72: # The file 'foo' should not be the same as 'bar' (0.000s)
+        > ! cmp foo bar
+        diff foo bar
+        --- foo
+        +++ bar
+        @@ -1,2 +1,1 @@
+        -foo
+        -
+        +bar
+
+    --- PASS: TestScript/example.txtar (0.00s)
+    ok      github.com/cilium/cilium/contrib/examples/script        0.003s
+
+In the test execution we can see that a temporary working directory ``$WORK`` was created
+and our test files from the ``example.txtar`` extracted there. Each command was then executed
+in order.
+
+As many of the cells bring rich set of commands it's important that they're easy to discover.
+This is helped by having a help system in place (``help``) and the ability to interactively try things out
+with the ``break`` statement. Here's what would happen if you'd add ``break`` as the last command in ``example.txtar``:
```

Comment:
```suggestion
As many of the cells bring rich set of commands it's important that they're easy to discover. To find the commands available, use the ``help`` command to interactively
explore the available operations to run as part of a test. Here's what would happen if you add ``break`` as the last command in ``example.txtar``:
```

Reply:
^ This is a general tone/voice suggestion for docs: Directly instruct the reader what do do. This is better than just informing them it's possible to do something, as the language is simpler and more direct.

Reply:
(I might also suggest even rewording "here's what would happen" to instead instruct the user to add the break command and then run the help command; I didn't spend much time rewording it, but even the existing suggestion is a slight improvement on the existing text IMO)

Reply:
Thanks, good suggestions. Applied.